### PR TITLE
PV2-1446_do_not_retry_on_contract_type_none

### PR DESCRIPTION
**Pull request by kuzinator**

### CODE REVIEW

File: src/SFA.DAS.Payments.EarningEvents.Application/Mapping/ApprenticeshipContractTypeEarningsEventFactory.cs [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
-        public const string ContractForServicesWithEmployer = "Contract for services with the employer";
-        public const string ContractForServicesWithSfa = "Contract for services with the ESFA";
-
         public ApprenticeshipContractTypeEarningsEvent Create(string contractType)
         {
             switch (contractType)
             {
-                case Act1:
-                case ContractForServicesWithEmployer:
+                case ContractTypeValidationRule.Act1:
```
Comment (carlbm) [inline src/SFA.DAS.Payments.EarningEvents.Application/Mapping/ApprenticeshipContractTypeEarningsEventFactory.cs]: I don't think that the `ContractTypeValidationRule` is the correct place for the ACT1/2 values. Maybe move them to their own class in `Payments.Model.Core`?

File: src/SFA.DAS.Payments.EarningEvents.Domain.UnitTests/Validation/Learner/ContractTypeValidationRuleTest.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                        PriceEpisodeIdentifier = "pe-1",
+                        PriceEpisodeValues = new PriceEpisodeValues
+                        {
+                            EpisodeStartDate = DateTime.Today.AddMonths(-2),
+                            PriceEpisodePlannedEndDate = DateTime.Today.AddDays(-30),
+                            PriceEpisodeActualEndDate = DateTime.Today.AddDays(-30),
+                            PriceEpisodeContractType = contractType
+                        }
+                    },
+                    new PriceEpisode
```
Comment (carlbm) [inline src/SFA.DAS.Payments.EarningEvents.Domain.UnitTests/Validation/Learner/ContractTypeValidationRuleTest.cs]: Do you need this?

File: src/SFA.DAS.Payments.EarningEvents.Application/Mapping/MappingExtensions.cs [hunk 3]
```diff
@@ -97,32 +96,5 @@ public static List<EarningPeriod> CreateIncentiveEarningPeriods(this PriceEpisod
             result.AddPeriodValue(values.Period12, 12, priceEpisodeIdentifier, 1);
             return result;
         }
-
-        public static PriceEpisode GetLatestOnProgPriceEpisode(this List<PriceEpisode> priceEpisodes)
```
Comment (carlbm) [inline src/SFA.DAS.Payments.EarningEvents.Application/Mapping/MappingExtensions.cs]: Were these methods just not used?

**Comment by carlbm** [comment]
Something to discuss - if there is a validation failure doesn't that just throw an exception that is handled by NServiceBus through the normal retry mechanism?

**Comment by benagain** [comment]
This approach is obsolete and has been replaced by #765

**Pull request closed without merging**